src/db.c

```diff
@@ -2559,8 +2566,9 @@ void db_col_transaction_commit(struct db_filter_col *col)
 	if (snap->shadow) {
 		/* leave the shadow intact, but drop the next snapshot */
```

Comment:
Is this block of code accessible?  I hacked around for just a bit and was unable to hit it.

`snap->shadow` is effectively always set to `false` at the start of `db_col_transaction_start()`.  `snap->shadow` is only set `true` at the end of `db_col_transaction_commit()` - approximately 100 lines after this.  Is there a code path where `snap->shadow` is `true` at this point?

Reply:
Unfortunately, it looks like I never responded back when I "fixed" this, so I'm sorta going by memory/comments/self-code-review and I believe the answer that yes, you can hit this code when you start nesting transactions.